ios/MaasCore/Sources/MaasComponents/Components/BottomSheet/BottomSheet.swift

```diff
@@ -0,0 +1,142 @@
+import SwiftUI
+import Combine
+
+struct BottomSheet: ViewModifier {
+
+    @Themeable(BottomSheetConstants.init) var constants
+
+    @EnvironmentObject var environment: BottomSheetEnvironment
+
+    @State var dragOffset: CGFloat = 0
+
+    @State var contentBounds: CGRect = .zero
+
+    @State var presenterBounds: CGRect = .zero
+
+    var topAnchor: CGFloat { presenterBounds.height - contentBounds.height }
+
+    var bottomAnchor: CGFloat { UIScreen.main.bounds.height }
+
+    var topInset: CGFloat {
+        UIApplication.shared.windows.first?.safeAreaInsets.top ?? 20.0
+    }
+
+    var sheetPosition: CGFloat {
+        if environment.isPresented {
+            let position = topAnchor + dragOffset
+            return position < topInset
+                ? topInset
+                : position
+        } else {
+            return bottomAnchor - dragOffset
+        }
+    }
+
+    func body(content: Content) -> some View {
+        ZStack {
+            content
+            sheetContentView
+        }
+        .animation(.easeInOut(duration: BottomSheetEnvironment.animationDuration))
+        .bounds(key: PresenterContentPreferenceKey.self) { presenterBounds = $0 }
+    }
+}
+
+// MARK: - Bottom Sheet Appearance
+
+private extension BottomSheet {
+
+    var sheetKnobView: some View {
+        RoundedRectangle(cornerRadius: 2.5)
+            .fill(constants.knobColor)
+            .frame(width: 32, height: 5, alignment: .center)
+            .padding(.top, 8)
+    }
+
+    var backgroundView: some View {
+        constants
+            .backgroundColor
+            .onTapGesture {
+                withAnimation(.easeInOut(duration: BottomSheetEnvironment.animationDuration)) {
+                    environment.isPresented = false
+                    environment.onDismiss?()
+                }
+            }
+            .edgesIgnoringSafeArea(.vertical)
+    }
+
+    var sheetContentView: some View {
+        ZStack {
+
+            Group {
+                if environment.isPresented {
+                    backgroundView
+                }
+            }
+
+            VStack(spacing: 0) {
+                sheetKnobView
+
+                VStack {
+
+                    environment.sheetContent
+                        .padding([.top, .bottom], constants.sheetContentVerticalPadding)
+                        .padding([.leading, .trailing], constants.sheetContentHorizontalPadding)
+                        .bounds(key: ContentPreferenceKey.self) { contentBounds = $0 }
+
+                    Spacer()
+                }
+            }
+            .frame(width: UIScreen.main.bounds.width)
+            .background(Color(.systemBackground))
```

Comment:
you have this in constants, no? 🤔 
```suggestion
            .background(constants.backgroundColor)
```